src/python/pants/backend/codegen/protobuf/java/dependency_inference.py

```diff
@@ -69,6 +76,54 @@ async def resolve_protobuf_java_runtime_for_resolve(
     return ProtobufJavaRuntimeForResolve(addresses)
 
 
+@dataclass(frozen=True)
+class ProtobufJavaGrpcRuntimeForResolveRequest:
+    resolve_name: str
+
+
+@dataclass(frozen=True)
+class ProtobufJavaGrpcRuntimeForResolve:
+    addresses: FrozenSet[Address]
+
+
+@rule
+async def resolve_protobuf_java_grpc_runtime_for_resolve(
+    jvm_artifact_targets: AllJvmArtifactTargets,
+    jvm: JvmSubsystem,
+    request: ProtobufJavaGrpcRuntimeForResolveRequest,
+) -> ProtobufJavaGrpcRuntimeForResolve:
+    addresses = find_jvm_artifacts_or_raise(
+        required_coordinates=[
+            # For non-Android uses:
+            # TODO: Maybe support Android jars? See https://github.com/grpc/grpc-java#download for
+            # the differences in required jars.
+            UnversionedCoordinate(
+                group="io.grpc",
+                artifact="grpc-netty-shaded",
+            ),
+            UnversionedCoordinate(
+                group="io.grpc",
+                artifact="grpc-protobuf",
+            ),
+            UnversionedCoordinate(
+                group="io.grpc",
+                artifact="grpc-stub",
+            ),
+            # TODO: This is only required for JDK 9+ according to https://github.com/grpc/grpc-java#download.
```

Comment:
Is anyone still using JDK 8? Hopefully not 7 lol. Maybe not worth keeping this TODO

Reply:
Possibly. I wouldn't put it past someone to still be on JDK8.

Reply:
I don't think it's worth Pants spending engineering time optimizing for those JDK 8 users. As long we don't _break things_ by having this extra req, I think we should remove this TODO. 

(Good TODO hygiene to only have things we actually want to do)

Reply:
8 is still very out there unfortunately.

Reply:
Will leave in TODO then.